### PR TITLE
Add integration tests covering all adapters

### PR DESCRIPTION
**Pull request by dmarcelino**
## Objectives
 * detect if a change in waterline core breaks the API;
 * detect if a change in waterline-adapter-tests breaks any official adapter tests;
 * test using the master versions of waterline core, waterline-adapter-tests and the adapters to ensure the current snapshot of all these are working together and consequently are OK to release;
 * make it easier for waterline core developers to test changes against all official adapters (it's not fun to setup all these dbs and run tests 5 times).

## What does this do
Runs `waterline-adapter-tests` against all official adapters.

## Build results:
Please check the below link for results:
https://travis-ci.org/dmarcelino/waterline-adapter-tests/jobs/55738380#L754

| adapter          | status  | # tests failing
|------------------|---------|----------
| sails-postgresql | failed  | 4
| sails-memory     | failed  | 2
| sails-disk       | failed  | 2
| sails-mongo      | failed  | 7
| sails-mysql      | failed  | 6

A bit scary, currently not a single official adapter is passing the integration tests when running their master branch version against waterline/waterline-adapter-tests master branches. We should make fixing this a priority!

## Feedback
Your feedback is welcomed. I'm quite open to making changes so let me know your opinions!

Note: I'm not sure if the best place for these integration tests is in this project or in waterline core. Initially I was afraid these tests would be slow but they actually run under a minute. Anyway, Waterline already has its own set of tests and adding the `sails-memory` tests to it (balderdashy/waterline#896) should be a good enough *canary test*. We also probably don't want to have broken waterline core builds because of an adapter that hasn't been updated recently. Open to suggestions.

cc: @tjwebb, @devinivy, @randallmeeker, @particlebanana

### CODE REVIEW

**Comment by tjwebb** [comment]
They are passing on Travis-CI, right? Can you clarify what the difference is here that makes them fail

**Comment by devinivy** [comment]
I am a little confused by this at first glance too.  For example, on sails-memory,
1. It passes CI.  200 passing / 0 failing / 2 pending
2. It fails on my machine.  19 passing / 1 failing / 0 pending (?!)
3. It fails on this PR . 201 passing / 2 failing / 2 pending

What am I missing here?

**Comment by dmarcelino** [comment]
@tjwebb, @devinivy, oops I should have explained myself better. So, the current adapter tests setup is good but has 2 deficiencies:

1. Travis builds only run when a change is pushed or a project owner presses the travis build button;
2. The adapters are only tested against the latest npm (not github) version of waterline and waterline-adapter-tests.

There is nothing wrong *per se* with this but it makes these tests little useful for waterline/waterline-adapter-tests core developers. @devinivy the reason for different result sets has probably got to do with different versions of waterline/waterline-adapter-tests. The table below gives you a breakdown of what the current adapters have been tested against and when:

| adapter | last build | waterline version | waterline-adapter-tests versions |
|------------|-------------|------------------------|---------------------------------------------|
| sails-postgresql | [12 days ago](https://travis-ci.org/balderdashy/sails-postgresql/builds/54277398) | 0.10.18 | 0.10.8  |
| sails-mysql | [7 days ago](https://travis-ci.org/balderdashy/sails-mysql/builds/54893696) | 0.10.18 | 0.10.8  |
| sails-mongo | [7 days ago](https://travis-ci.org/balderdashy/sails-mongo/builds/54907059) | 0.10.18 | 0.10.8  |
| sails-memory | [7 months ago](https://travis-ci.org/balderdashy/sails-memory/builds/33848730) | 0.10.9 | 0.10.4  |
| sails-disk | [3 months ago](https://travis-ci.org/balderdashy/sails-disk/builds/45490139) | 0.10.14 | 0.10.5  |

waterline / waterline adapter status:

 project | last commit | commits since last release* 
-----------|-----------------|--------------------------------------
 waterline | 4 days ago | [25 commits](https://github.com/balderdashy/waterline/compare/v0.10.18...master) 
 waterline-adapter-tests | 16 hours ago | [2 commits](https://github.com/balderdashy/waterline-adapter-tests/compare/v0.10.8...master) 

*AKA commits untested against latest version of adapters

As you can see from the above tables, each adapter set of tests never ran against the latests commits in waterline core. Ironically the more stable an adapter is the more likely it is for a change in waterline core to break it. Look at sails-memory and sails-disk, because they have been so stable they haven't ran for months!

PS: In addition to my PR we should also look into implementing https://github.com/FiloSottile/travis-cron or https://github.com/grosser/travis_cron for waterline, waterline-adapter-tests and all official adapters.

**Comment by devinivy** [comment]
Indeed!  All good points.  In my case, it was only running 20 tests because mocha has `bail: true` in the sails-memory config.  Once I commented that out, I got the same results as in this PR.

TL;DR for @dmarcelino's comment (but do read it!): those adapters **are _legitimately_ failing** CI right now due to updates to waterline-adapter-test and waterline since their last build.  That realization makes this PR yet more crucial, and it answers @tjwebb's original question.

Luckily most of the failing tests seem reasonably easy to fix.  Can we push this PR through?  The code looks good to me.

**Comment by particlebanana** [comment]
This is awesome! Go for it!

**Comment by dmarcelino** [comment]
Thanks @particlebanana.

**Comment by devinivy** [comment]
Ahhhhhh :)

**Comment by dmarcelino** [comment]
@devinivy now we "only" need to get those tests passing :wink:
